From 457a9644c0581d033231519fe31b6f7ecd75fc02 Mon Sep 17 00:00:00 2001
From: DelphineCCChiu <delphine_cc_chiu@wiwynn.com>
Date: Tue, 23 Apr 2024 13:57:49 +0800
Subject: [PATCH] platform-mc: Monitor all sensors once upon the first power-on

BMC reads all sensors once upon reboot, then update sensor values according to the update interval time.

Change-Id: I489bf370dfa9b511ad3004431504ddc96c3ac0a4
Signed-off-by: DelphineCCChiu <delphine_cc_chiu@wiwynn.com>
---
 platform-mc/sensor_manager.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/platform-mc/sensor_manager.cpp b/platform-mc/sensor_manager.cpp
index 4ea734d..8bbb58f 100644
--- a/platform-mc/sensor_manager.cpp
+++ b/platform-mc/sensor_manager.cpp
@@ -222,7 +222,7 @@ requester::Coroutine SensorManager::doSensorPollingTask(pldm_tid_t tid)
 
             sd_event_now(event.get(), CLOCK_MONOTONIC, &t1);
             elapsed = t1 - sensor->timeStamp;
-            if (sensor->updateTime <= elapsed)
+            if ((sensor->updateTime <= elapsed) || (sensor->timeStamp == 0))
             {
                 co_await getSensorReading(sensor);
                 if (sensorPollTimers[tid] &&
-- 
2.25.1

